manager/network/model.go

```diff
@@ -0,0 +1,15 @@
+package network
+
+import (
+	"github.com/docker/swarmkit/api"
+	"github.com/docker/swarmkit/manager/allocator/networkallocator"
+)
+
+// Model is an abstraction over the Network Model to be used.
+type Model interface {
+	Allocator() (networkallocator.NetworkAllocator, error)
+	SetDefaults(spec *api.NetworkSpec) error
+	ValidateNetworkSpec(spec *api.NetworkSpec) error
+	PredefinedNetworks() []networkallocator.PredefinedNetworkData
+	SupportIngressNetwork() bool
```

Comment:
Another way to do this is only define an `IngressNetwork` method on implementations that support the ingress network, and type-assert to check whether a particular implementation has this method or not. This keeps implementation-specific details from creeping into the main interface definition. It can be more cumbersome to use, though.